experimental/sql/dialects/dataflow.py

```diff
@@ -0,0 +1,113 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from typing import Any, Union, List
+from xdsl.ir import Block, Region, Operation, SSAValue, ParametrizedAttribute, Data, MLContext, Attribute
+from xdsl.dialects.builtin import StringAttr, ArrayAttr, ArrayOfConstraint, IntegerAttr, IntegerType
+from xdsl.irdl import AttributeDef, OperandDef, ResultDef, RegionDef, SingleBlockRegionDef, irdl_attr_definition, irdl_op_definition, ParameterDef, AnyAttr, VarOperandDef, builder
+
+#===------------------------------------------------------------------------===#
+# Data types
+#===------------------------------------------------------------------------===#
+
+#===------------------------------------------------------------------------===#
+# Query types
+#===------------------------------------------------------------------------===#
+
+
+@irdl_attr_definition
+class Tuple(ParametrizedAttribute):
```

Comment:
AFAIK, you did not specify the tuple type in this dialect. Is the tuple you are using some builtin thing from LLVM, @ingomueller-net?

In that case, I would probably have t add it to xdsl too, s.t. we can actually lower.

Reply:
There is actually a [`TupleType`](https://mlir.llvm.org/docs/Dialects/Builtin/#tupletype) in the "built-in" dialect (defined [here](https://github.com/llvm/llvm-project/blob/3bacd94c312c35684ac5985d37c594acb62b2aa6/mlir/include/mlir/IR/BuiltinTypes.td#L739)).